estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,59 @@
+from datetime import timedelta
+
+from odoo import api, fields, models
+
+
+class EstatePropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "Estate Property Offer"
+
+    price = fields.Float("Price")
+    status = fields.Selection(
+        selection=[
+            ("accepted", "Accepted"),
+            ("refused", "Refused"),
+        ],
+        string="Status",
+        copy=False,
+    )
+    partner_id = fields.Many2one("res.partner", string="Partner", required=True)
+    property_id = fields.Many2one("estate.property", string="Property", required=True)
+    validity = fields.Integer("Validity (days)", default=7)
+    date_deadline = fields.Date("Deadline", compute="_compute_date_deadline", inverse="_inverse_date_deadline")
+
+    @api.depends("create_date", "validity")
+    def _compute_date_deadline(self):
+        for offer in self:
+            if offer.create_date:
+                offer.date_deadline = offer.create_date.date() + timedelta(days=offer.validity)
+            else:  # Fallback if create_date is not set
+                offer.date_deadline = fields.Date.today() + timedelta(days=offer.validity)
+
+    def _inverse_date_deadline(self):
+        for offer in self:
+            if offer.date_deadline and offer.create_date:
+                offer.validity = (offer.date_deadline - offer.create_date.date()).days
+            elif offer.date_deadline:  # Fallback if create_date is not set
+                offer.validity = (offer.date_deadline - fields.Date.today()).days
+
+    def action_accept_offer(self):
+        for offer in self:
+            offer.status = "accepted"
+            # Refuse other offers for the same property
+            other_offers = self.search(
+                [
+                    ("property_id", "=", offer.property_id.id),
+                    ("id", "!=", offer.id),
+                ]
+            )
```

Comment:
it's good, but you can make it a bit simpler. Recordsets have bitwise operations (`__sub__`, `__or__`, `__and__` etc)
```suggestion
            other_offers = self.property_id.offer_ids - offer
```

Also check that none other offers were already accepted. e.g.: 
```py
if any(other_offers.filtered(lambda o: o.status == 'accepted')): ...
```

Reply:
thx! fixed.